packages/account/src/Sections/Verification/ProofOfIncome/proof-of-income-form.jsx

```diff
@@ -0,0 +1,221 @@
+import PropTypes from 'prop-types';
+import {
+    Autocomplete,
+    Button,
+    DesktopWrapper,
+    MobileWrapper,
+    SelectNative,
+    FormSubmitErrorMessage,
+    Text,
+    Timeline,
+} from '@deriv/components';
+import { Formik, Field } from 'formik';
+import FormSubHeader from 'Components/form-sub-header';
+import { localize, Localize } from '@deriv/translations';
+import { WS } from '@deriv/shared';
+import { connect } from 'Stores/connect';
+import FormFooter from 'Components/form-footer';
+import LoadErrorMessage from 'Components/load-error-message';
+import PoincFileUploaderContainer from 'Components/poinc/file-uploader-container';
+
+let file_uploader_ref = null;
+
+const ProofOfIncomeForm = ({
+    addNotificationByKey,
+    removeNotificationByKey,
+    removeNotificationMessage,
+    poinc_documents_list,
+    onSubmit,
+}) => {
+    const [document_file, setDocumentFile] = React.useState({ files: [], error_message: null });
+    const [disabled_items, setDisabledItems] = React.useState([2]);
+    const [api_initial_load_error, setAPIInitialLoadError] = React.useState(null);
+    const [uploading_document_type, setUploadingDocumentType] = React.useState('');
+
+    const initial_form_values = {
+        document_type: '',
+    };
+
+    const validateFields = values => {
+        const errors = {};
+        const { document_type } = values;
+
+        if (!document_type) {
+            errors.document_type = localize('This field is required.');
+        } else if (!poinc_documents_list.find(c => c.text === document_type)) {
+            errors.document_type = localize('This field is required.');
+        }
+
+        if (!errors.document_type) {
+            setDisabledItems([]);
+        } else if (!disabled_items.includes(2)) {
+            setDisabledItems([2]);
+        }
+
+        return errors;
+    };
+
+    const onSubmitValues = (values, { setStatus, setSubmitting }) => {
+        const uploading_value = poinc_documents_list.find(doc => doc.text === values.document_type)?.value;
+        setUploadingDocumentType(uploading_value);
+        setStatus({ msg: '' });
+
+        file_uploader_ref?.current
+            .upload()
+            .then(api_response => {
+                if (api_response.warning) {
+                    setStatus({ msg: api_response.message });
+                } else {
+                    WS.authorized.storage.getAccountStatus().then(({ error, get_account_status }) => {
+                        if (error) {
+                            setAPIInitialLoadError(error.message);
+                            return;
+                        }
+                        const { income, needs_verification } = get_account_status.authentication;
+                        const needs_poinc =
+                            needs_verification.includes('income') && ['rejected', 'none'].includes(income?.status);
+                        removeNotificationMessage({ key: 'authenticate' });
```

Comment:
Why do we have both? - `removeNotificationMessage` and `removeNotificationByKey` ?

Reply:
they are different functions. You can check in notification-store. 
this part of the code is similar to proof of address form